CometServer/Modules/Authentication/UsernameModule.cs

```diff
@@ -43,25 +45,10 @@ public class UsernameModule : CarterModule
         /// </param>
         public override void AddRoutes(IEndpointRouteBuilder app)
         {
-            app.MapGet("/username", async (HttpRequest req, HttpResponse res) => {
-                if (!req.HttpContext.User.Identity.IsAuthenticated)
-                {
-                    res.UpdateWithNotAuthenticatedSettings();
-
-                    await res.WriteAsync("not authenticated");
-                }
-                else
-                {
-                    if (res.HttpContext.User == null)
-                    {
-                        res.UpdateWithNotAutherizedSettings();
-                        
-                        await res.WriteAsync("not authorized");
-                    }
-
-                    await res.WriteAsync(res.HttpContext.User.Identity.Name);
-                }
-            });
+            app.MapGet("/username", async (HttpRequest req, HttpResponse res) =>
+            {
+                await res.WriteAsync(res.HttpContext.User.Identity.Name);
+            }).RequireAuthorization(new[] { BasicAuthenticationDefaults.AuthenticationScheme });
```

Comment:
does this only work with basic then?

Reply:
Indeed, I support everything is necessary?